modules/webhook/webhook.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package webhook
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+)
+
+var Webhooks = make(map[string]*Webhook)
+
+// Webhook is a custom webhook
+type Webhook struct {
+	ID    string   `yaml:"id"`
+	Label string   `yaml:"label"`
+	URL   string   `yaml:"url"`
```

Comment:
I wonder if `docs` would be better instead of `url`. `url` could mean anything really and would definitely require a lookup in the gitea docs. Meanwhile, `docs` is more self-explanatory.